juriscraper/opinions/united_states/federal_bankruptcy/bap1.py

```diff
@@ -0,0 +1,65 @@
+"""
+Scraper for the United States Bankruptcy Appellate Panel for the First Circuit
+CourtID: bap1
+Court Short Name: 1st Cir. BAP
+"""
+
+from lxml.html import HtmlElement
+
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.string_utils import titlecase
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+
+
+class Site(OpinionSiteLinear):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.url = "https://www.bap1.uscourts.gov/bapopn"
+        self.court_id = self.__module__
+
+    def _download(self, request_dict={}) -> None:
+        html_tree = super()._download(request_dict)
+        self.extract_cases(html_tree)
+
+        if self.test_mode_enabled():
+            return html_tree
+
+        # Walk over pagination "Next" page(s), if present
+        proceed = True
+
+        while proceed:
+            next_page_url = self.extract_next_page_url(html_tree)
+
+            if next_page_url:
+                logger.info(f"Scraping next page: {next_page_url}")
+                html_tree = self._get_html_tree_by_url(next_page_url)
+                self.extract_cases(html_tree)
+            else:
+                proceed = False
```

Comment:
download doesn't need to do any pagination - that all occurs in the backscrape with the iterable.  I may have sent you the wrong example but here is how I would do back scrape iterable 

`self.back_scrape_iterable = list(range(1, 29))[::-1]`